server/import.go

```diff
@@ -0,0 +1,192 @@
+package server
+
+import (
+	"crypto"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/rsa"
+	"github.com/google/go-tpm-tools/tpm2tools"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+	"hash"
+	"io"
+
+	pb "github.com/google/go-tpm-tools/proto"
+)
+
+// hashConstants contains all the per-EK constant hash values in a convenient struct.
+type hashConstants struct {
```

Comment:
This seems unnecessary,

We should just pass `ek.NameAlg` (a `tpm2.Algorithm`) everywhere and use the `HashConstructor` and `Size` methods when necessary.

Right now we aren't consistent. Sometimes we pass a `hash.Hash` sometimes a `hashConstants`, sometimes a `tpm2.Algorithm`. If you want things to be more readable a helper functions like:

```go
func getHash(alg tpm2.Algorithm) hash.Hash {
	create, err := alg.HashConstructor()
	if err != nil {
		panic(err)
	}
	return create()
}
```

might be useful. Then you can just pass `alg` everywhere and use `alg`, `getHash(alg)`, and `getHash(alg).Size()`